File: 20. Valid Parentheses/20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        """
        :type s: str
        :rtype: boo.
        """
        
        parentheses = {
            "(" : ")",
            "[" : "]",
            "{" : "}"
        }
        
        stack = []
        for c in s: 
            if c in parentheses : 
                stack.append(c)
            else : 
                if len(stack) == 0 or parentheses[stack.pop()] != c : 
                    return False
        
        return len(stack) == 0

File: 20. Valid Parentheses/20. Valid Parentheses.js
var isValid = function(s) {
    const parentheses = {
        "(" : ")",
        "[" : "]",
        "{" : "}"
    }

    let stack = []

    for (c of s){
        if (Object.keys(parentheses).includes(c)){
            stack.push(c)
        }
        else {
            if (stack.length === 0 || parentheses[stack.pop()] !== c){
                return false
            }
        }
    }

    return stack.length === 0 ? true : false
}